### PR TITLE
Fixed windows build

### LINKED ISSUE #2: Windows cmake 3.21.4 build fails
Hello,

When compiling the Anari SDK on Windows 11 Visual Studio 2019 I get this build error:

`8>Generating Code...
8>Done building project "anari_library_example.vcxproj" -- FAILED.
12>------ Build started: Project: anariTutorialDirectLink, Configuration: Release x64 ------
12>Building Custom Rule C:/Users/Cameron/Downloads/ANARI-SDK/examples/direct_linking/CMakeLists.txt
12>anariTutorialDirectLink.c
12>LINK : fatal error LNK1181: cannot open input file '..\..\Release\anari_library_example.lib'
12>Done building project "anariTutorialDirectLink.vcxproj" -- FAILED.
13>------ Build started: Project: ALL_BUILD, Configuration: Release x64 ------
13>Building Custom Rule C:/Users/Cameron/Downloads/ANARI-SDK/CMakeLists.txt
========== Build: 8 succeeded, 5 failed, 0 up-to-date, 0 skipped ==========
`
I attempt to continue the build:

`1>C:\Users\Cameron\Downloads\ANARI-SDK\examples\example_device\scene\bvh/SweepSAHBuilder.h(294,3): error C2059: syntax error: '{'
1>C:\Users\Cameron\Downloads\ANARI-SDK\examples\example_device\scene\bvh/SweepSAHBuilder.h(294,3): error C2143: syntax error: missing ';' before '{'
1>C:\Users\Cameron\Downloads\ANARI-SDK\examples\example_device\scene\bvh/SweepSAHBuilder.h(301,31): warning C4267: '=': conversion from 'size_t' to 'uint32_t', possible loss of data
1>C:\Users\Cameron\Downloads\ANARI-SDK\examples\example_device\scene\volume\../SceneObject.h(84,9): error C3001: 'declare': expected an OpenMP directive name
1>C:\Users\Cameron\Downloads\ANARI-SDK\examples\example_device\scene\volume\../SceneObject.h(88,9): error C3036: 'bbox_extend': invalid operator token in OpenMP 'reduction' clause
1>C:\Users\Cameron\Downloads\ANARI-SDK\examples\example_device\scene\volume\../SceneObject.h(88,1): error C3016: 'i': index variable in OpenMP 'for' statement must have signed integral type
1>Generating Code...
1>Done building project "anari_library_example.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 3 up-to-date, 0 skipped ==========`

Am I missing prerequisites? I hope this is helpful.

Kind regards,

Cameron

### PR DESCRIPTION
**Pull request by LDeakin**
This fixes all the issues I identified when trying to build the ANARI SDK in a Windows environment, and should close #2.
Most of the issues were related to symbol exporting.

### CODE REVIEW

**Comment by johguenther** [comment]
Thanks! Looks good, just some minor suggestions.

**Comment by LDeakin** [comment]
Thanks Johannes, I've resolved those issues you identified.

I identified a few other issues when trying to build with `BUILD_SHARED_LIBS=OFF`. I've made some additional changes in that commit, so it builds. However, the dynamically linked executables fail at runtime.

The issue is that when the device is registered (see [here](https://github.com/KhronosGroup/ANARI-SDK/blob/ffe8f2050ea0520751db546eea2a6fbd401964d9/examples/example_device/ExampleDevice.cpp#L534-L538)), `anari::device::registerType` registers the device type in a static variable in the DLL. However, `anariNewDevice` is calling `anari::Device::createDevice` which is using the executables own copy of that static variable, where the device type is not registered.
I'm not sure what the best way to address this is.

**Comment by jeffamstutz** [comment]
I think the right solution is to have `anari::Library` store the library's creation function map, instead of it be statically stored globally in the based `anari::Device` class. This would the privatize all device creation to a concrete instance of `Library`.

**Comment by jeffamstutz** [comment]
FYI I'd be happy to implement my idea tomorrow. @LDeakin would you be ok with rebasing on `main` after I do that and incorporate it into your PR?

**Comment by jeffamstutz** [comment]
Sorry this took so long, but I implemented it in 2e57ee4. All static symbols in `anari::Device` are gone.

**Comment by LDeakin** [comment]
Thanks @jeffamstutz , I've rebased on main. Seems like everything is working as expected on Linux and Windows now.

File: examples/example_device/CMakeLists.txt [hunk 1]
```diff
@@ -89,6 +89,7 @@ endif()
 target_compile_definitions(${PROJECT_NAME}
 PRIVATE
   -DDEVICE_VERSION=${REFERENCE_DEVICE_VERSION}
+  "example_device_EXPORTS"
```
Comment (johguenther) [inline examples/example_device/CMakeLists.txt:92]: Interesting: I didn't know that leading `-D` are removed anyway...

**Comment by johguenther** [comment]
Thanks!

**Pull request merged**